superset-frontend/packages/superset-ui-core/src/query/types/QueryFormData.ts

```diff
@@ -146,7 +147,7 @@ export type ExtraFormData = ExtraFormDataAppend & ExtraFormDataOverride;
 
 export interface BaseFormData extends TimeRange, FormDataResidual {
   /** datasource identifier ${id}_${type} */
-  datasource: string;
+  datasource: Datasource;
```

Comment:
TL;DR: this should be unchanged.

I believe this should remain as `string` - this will change in Superset 2.0, but right now `datasource` in the chart metadata is stored as a string, e.g. `1__table`, which represents the primary keys of a base datasource object in the backend. That can then be fetched from the backend, and when that's done, it then becomes a full `Datasource` object, something like
```javascript
{
  id: 1,
  type: 'string',
  name: 'my_table',
  ...
}
```
which is later returned in `chartProps`.

Reply:
Thanks @villebro. Found the issue and was successfully able to resolve it.
Did the following: 
1. Reverted the Type change and applied the fix
3. Ran Lint on the `superset-frontend` folder
4. Retested the UI after changes

![res](https://user-images.githubusercontent.com/42173629/145186221-b06cc323-cdf1-46b4-829e-7c865a077571.PNG)